core/Pos/Core/Address.hs

```diff
@@ -83,10 +82,8 @@ instance Bi Address => Hashable AddressIgnoringAttributes where
 decodeAddress :: Bi Address => ByteString -> Either String Address
 decodeAddress bs = do
     let base58Err = "Invalid base58 representation of address"
-        takeErr = toString . view _3
-        takeRes = view _3
     dbs <- maybeToRight base58Err $ decodeBase58 addrAlphabet bs
-    bimap takeErr takeRes $ Bi.decodeOrFail $ BSL.fromStrict dbs
+    bimap toString identity $ Bi.decodeFull dbs
```

Comment:
Though here I'd rather use `over _Left` because it took me some time to understand that `first` works for `Either` as well.